### PR TITLE
Prm 3176 removed obligation joruney

### PR DESCRIPTION
**Pull request by RoheelArshad**

### CODE REVIEW

File: app/controllers/HonestyDeclarationController.scala [hunk 1]
```diff
@@ -95,11 +94,10 @@ class HonestyDeclarationController @Inject()(honestDeclarationPage: HonestyDecla
     Future[Result])(implicit request: UserRequest[_]): Future[Result] = {
     (request.answers.getAnswer[String](SessionKeys.reasonableExcuse), request.answers.getAnswer[LocalDate](SessionKeys.dueDateOfPeriod),
       request.answers.getAnswer[LocalDate](SessionKeys.startDateOfPeriod), request.answers.getAnswer[LocalDate](SessionKeys.endDateOfPeriod),
-      request.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isDefined) match {
+      request.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isDefined) match {
       case (Some(reasonableExcuse), Some(dueDate), Some(startDate), Some(endDate), isObligation: Boolean) =>
```
Comment (AndrewRButcher) [inline app/controllers/HonestyDeclarationController.scala:98]: this param should not be needed anymore (only users with a reasonable excuse can access the honesty declaration page
```suggestion
      ) match {
      case (Some(reasonableExcuse), Some(dueDate), Some(startDate), Some(endDate)) =>
```

File: app/controllers/ViewAppealDetailsController.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ class ViewAppealDetailsController @Inject()(viewAppealDetailsPage: ViewAppealDet
                     implicit val userRequest: UserRequest[AnyContent] = UserRequest(request.vrn, request.active, request.arn, userAnswers)(request)
                     val reasonableExcuseSpecificAnswers = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage()
                     val appealMetaInformation = sessionAnswersHelper.getSubmittedAnswers(getFeatureDate)
-                    if (userAnswers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isDefined || userRequest.answers.getAnswer[String](SessionKeys.reasonableExcuse).contains("other")) {
+                    if (userAnswers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isDefined || userRequest.answers.getAnswer[String](SessionKeys.reasonableExcuse).contains("other")) {
```
Comment (AndrewRButcher) [inline app/controllers/ViewAppealDetailsController.scala:66]:
```suggestion
                    if (userRequest.answers.getAnswer[String](SessionKeys.reasonableExcuse).contains("other")) {
```

File: app/controllers/YouCanAppealPenaltyController.scala [hunk 3]
```diff
@@ -17,7 +17,7 @@
 package controllers
```
Comment (AndrewRButcher) [inline app/controllers/YouCanAppealPenaltyController.scala:17]: I believe this controller and appropriate view view can be removed

File: app/helpers/IsLateAppealHelper.scala [hunk 4]
```diff
@@ -27,7 +27,7 @@ class IsLateAppealHelper @Inject()(dateTimeHelper: DateTimeHelper,
                                     appConfig: AppConfig) {
   def isAppealLate()(implicit userRequest: UserRequest[_]): Boolean = {
     val dateNow: LocalDate = dateTimeHelper.dateNow
-    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isEmpty){
+    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isEmpty){
```
Comment (AndrewRButcher) [inline app/helpers/IsLateAppealHelper.scala:30]: Remove Line 30 & 44 -46

File: app/helpers/SessionAnswersHelper.scala [hunk 5]
```diff
@@ -453,7 +446,7 @@ class SessionAnswersHelper @Inject()(uploadJourneyRepository: UploadJourneyRepos
   }
 
   def getContentWithExistingUploadFileNames(reasonableExcuse: String)(implicit userRequest: UserRequest[_], messages: Messages): Future[Seq[QuestionAnswerRow]] = {
-    if (!reasonableExcuse.equals("other") && !userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).contains(true)) {
+    if (!reasonableExcuse.equals("other") && !userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).contains(true)) {
```
Comment (AndrewRButcher) [inline app/helpers/SessionAnswersHelper.scala:449]:
```suggestion
    if (!reasonableExcuse.equals("other")) {
```

File: app/navigation/Navigation.scala [hunk 6]
```diff
@@ -242,25 +239,23 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
...
-      case _ => throw new MatchError(s"[Navigation][getNextURLBasedOnReasonableExcuse] - Unknown reasonable excuse $reasonableExcuse")
+
     }
   }
 
   protected[navigation] def routeToMakingALateAppealOrCYAPage(userRequest: UserRequest[_], mode: Mode): Call = {
     if (isLateAppealHelper.isAppealLate()(userRequest)
       && (userRequest.answers.getAnswer[String](SessionKeys.lateAppealReason).isEmpty || mode == NormalMode)
-      && userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isEmpty) {
+      && userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isEmpty) {
```
Comment (AndrewRButcher) [inline app/navigation/Navigation.scala:258]:
```suggestion
      ) {
```

File: app/navigation/Navigation.scala [hunk 7]
```diff
@@ -361,31 +356,28 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
   }
 
   protected[navigation] def reverseRouteForAppealStartPage(userRequest: UserRequest[_]): Call = {
-    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isDefined) {
+    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isDefined) {
       routes.YouCanAppealPenaltyController.onPageLoad()
     } else {
       Call("GET", appConfig.penaltiesFrontendUrl)
     }
```
Comment (AndrewRButcher) [inline app/navigation/Navigation.scala:363]:
```suggestion
      Call("GET", appConfig.penaltiesFrontendUrl)
```

File: app/navigation/Navigation.scala [hunk 8]
```diff
@@ -361,31 +356,28 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
...
   }
 
   protected[navigation] def reverseRouteForHonestyDeclaration(userRequest: UserRequest[_]): Call = {
-    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isDefined) {
+    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isDefined) {
       routes.AppealStartController.onPageLoad()
     } else {
       routes.ReasonableExcuseController.onPageLoad()
     }
```
Comment (AndrewRButcher) [inline app/navigation/Navigation.scala:371]:
```suggestion
      routes.ReasonableExcuseController.onPageLoad()
```

File: app/navigation/Navigation.scala [hunk 9]
```diff
@@ -361,31 +356,28 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
...
       routes.OtherReasonController.onPageLoadForWhyReturnSubmittedLate(mode)
-    }
+
   }
 
   protected[navigation] def reverseRouteForMakingALateAppealPage(userRequest: UserRequest[_], mode: Mode, jsEnabled: Boolean): Call = {
-    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isDefined) {
+    if (userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isDefined) {
       reverseRoutingForUpload(userRequest, mode, jsEnabled)
     } else {
```
Comment (AndrewRButcher) [inline app/navigation/Navigation.scala:382]: remove lines 300 to 302 and 392

File: app/navigation/Navigation.scala [hunk 10]
```diff
@@ -437,7 +429,7 @@ class Navigation @Inject()(dateTimeHelper: DateTimeHelper,
     val dateNow: LocalDate = dateTimeHelper.dateNow
     if (dateSentParsed.isBefore(dateNow.minusDays(daysResultingInLateAppeal))
       && (userRequest.answers.getAnswer[String](SessionKeys.lateAppealReason).isEmpty || mode == NormalMode)
-      && userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).isEmpty) {
+      && userRequest.answers.getAnswer[Boolean](SessionKeys.isFindOutHowToAppeal).isEmpty) {
```
Comment (AndrewRButcher) [inline app/navigation/Navigation.scala:432]:
```suggestion
      ) {
```

**Review by AndrewRButcher — CHANGES_REQUESTED** [review]
Couple of extra comments:
1. Can any views remaining after amendments in `views.obligation` be moved to `views.findOutHowToAppeal`
2. ObligationAppealInformation model (and any associated uses)can be removed

File: app/navigation/Navigation.scala (hunk 9 above)
Comment (AndrewRButcher) [reply app/navigation/Navigation.scala:382]: Essentially yes (the line numbers maybe off but the `if (isObligationAppeal/isFindOutHowToAppeal...` part is no longer required only users on an reasonable excuse journey can access the late appeal page so no need for the if statement

**Review by JBlacklaws — APPROVED** [review]
LGTM

**Pull request merged**